src/main/java/org/moltimate/moltimatebackend/service/AlignmentService.java

```diff
@@ -131,47 +134,50 @@ private Alignment alignActiveSites(Structure structure, Motif motif) {
         ArrayList<Residue> activeSiteOutput = new ArrayList<>();
         Set<Residue> used = new HashSet<>();
 
-        for (int i = 0; i < seq1.size(); i++) {
-            activeSiteOutput.add(seq1.get(i));
-            used.add(seq1.get(i));
+        for (int i = 0; i < activeSiteResidueList.size(); i++) {
+            activeSiteOutput.add(activeSiteResidueList.get(i));
+            used.add(activeSiteResidueList.get(i));
         }
 
-        int missingResidues = 0;
         for (Residue residue : motif.getActiveSiteResidues()) {
             if (!used.contains(residue)) {
                 activeSiteOutput.add(residue);
-                missingResidues++;
             }
         }
 
-        List<Group> seq2Sorted = new ArrayList<>();
-        seq2Sorted.addAll(seq2);
-        Collections.sort(seq2Sorted, Comparator.comparingInt(o -> o.getResidueNumber()
+        List<Group> alignedResidueListSorted = new ArrayList<>();
+        alignedResidueListSorted.addAll(alignedResidueList);
+        alignedResidueListSorted.sort(Comparator.comparingInt(o -> o.getResidueNumber()
                 .getSeqNum()));
 
-        String alignmentString = AlignmentUtils.groupListToResString(seq2Sorted);
+        String alignmentString = AlignmentUtils.groupListToResString(alignedResidueListSorted);
         String motifResString = AlignmentUtils.residueListToResString(motif.getActiveSiteResidues());
         int distance = AlignmentUtils.levensteinDistance(alignmentString, motifResString);
 
-        if (seq2.size() > 1 && acceptableDistance(activeSiteOutput.size(), distance)) {
-            HashSet<Group> residues = new HashSet<>();
-            residueMap.values()
-                    .forEach(residues::addAll);
+        if (alignedResidueList.size() > 1 && acceptableDistance(activeSiteResidueList.size(), distance)) {
             Alignment alignment = new Alignment();
-            alignment.setActiveSiteResidues(activeSiteOutput);
+            alignment.setActiveSiteResidues(activeSiteResidueList);
             alignment.setMotifPdbId(motif.getPdbId());
             alignment.setMinDistance(distance);
             alignment.setMaxDistance(distance);
-            alignment.setAlignedResidues(seq2.stream()
+            alignment.setAlignedResidues(alignedResidueList.stream()
                                                  .map(Residue::fromGroup)
                                                  .collect(Collectors.toList()));
-            alignment.setRmsd(rmsd(motif.getPdbId(), motif.getActiveSiteResidues(), seq2));
+            alignment.setRmsd(rmsd(motif.getPdbId(), motif.getActiveSiteResidues(), alignedResidueList));
```

Comment:
We discussed at one point having `rmsd()` take in the structure instead of the PDB id and getting the structure from the PDB, so that we can call the PDB fewer times. 

Do we still have access to the structure with these changes?

Reply:
Good point, adding now